### ISSUE TITLE
empty_count false positive on hex values

### ISSUE THREAD

**Issue by alejandro-isaza** [realm/SwiftLint#2423]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following code causes a `empty_count` false positive:

```
if result.count <= 0xff {
    ...
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
abc.swift:115:30: error: Empty Count Violation: Prefer checking `isEmpty` over comparing `count` to zero. (empty_count)
```

### Environment

* SwiftLint version: 0.27.0
* Installation method used: Homebrew

* Which Xcode version: Xcode 10
* Sample: `echo "if result.count <= 0xff {}" | swiftlint lint --no-cache --use-stdin --enable-all-rules`

```swift
// This triggers a violation:
if result.count <= 0xff {
    ...
}
```
